ui/components/MesheryPlayComponent.js

```diff
@@ -75,54 +75,48 @@ const styles = (theme) => ({
   },
 });
 
-class MesheryPlayComponent extends React.Component {
-  constructor(props) {
-    super(props);
+function MesheryPlayComponent(props) {
+  const { classes, meshAdapters, setAdapter } = props
+  const router = useRouter();
+  const [adapter, setAdapterState] = useState(meshAdapters && meshAdapters.size > 0 ? meshAdapters[0] : {});
+  const prevMeshAdaptersRef = useRef();
 
-    const { meshAdapters } = props;
-    let adapter = {};
-    if (meshAdapters && meshAdapters.size > 0) {
-      adapter = meshAdapters[0];
-    }
-    this.state = {
-      adapter,
-    };
-  }
 
-  handleRouteChange = () => {
-    const queryParam = this.props?.router?.query?.adapter;
+  useEffect(() => {
+    prevMeshAdaptersRef.current = meshAdapters;
+  });
+
+  const prevMeshAdapters = prevMeshAdaptersRef.current;
+
+  const handleRouteChange = useCallback(() => {
+    const queryParam = router?.query?.adapter;
     if (queryParam) {
-      const selectedAdapter = this.props.meshAdapters.find(({ adapter_location }) => adapter_location === queryParam);
+      const selectedAdapter = meshAdapters.find(({ adapter_location }) => adapter_location === queryParam);
       if (selectedAdapter) {
-        this.setState({ adapter : selectedAdapter });
+        setAdapterState(selectedAdapter);
       }
-    } else if (this.props.meshAdapters.size > 0) {
-      this.setState({ adapter : this.props.meshAdapters.get(0) });
+    } else if (meshAdapters.size > 0) {
+      setAdapterState(meshAdapters.get(0));
     }
-  };
+  }, [router, meshAdapters]);
 
-  componentDidMount() {
-    const { router } = this.props;
-    router.events.on("routeChangeComplete", this.handleRouteChange);
-  }
+  useEffect(( ) => {
+    router.events.on("routeChangeComplete", handleRouteChange);
 
-  componentDidUpdate(prevProps) {
-    // update the adapter when the meshadapters props are changed
-    if (prevProps.meshAdapters?.size !== this.props.meshAdapters?.size && this.props.meshAdapters.size > 0) {
-      this.handleRouteChange();
+    if (prevMeshAdapters?.size !== meshAdapters?.size && meshAdapters.size > 0) {
+      handleRouteChange();
     }
-  }
 
-  componentWillUnmount() {
-    this.props.router.events.off("routeChangeComplete", this.handleRouteChange);
-  }
+    return () => {
+      router.events.off("routeChangeComplete", handleRouteChange);
+    };
+  }, [router, meshAdapters]); // Dependencies for useEffect
```

Comment:
Thanks for adding comment but I think this comment is not required
```suggestion
  }, [router, meshAdapters]);
```